Feature/psalm dependency

william-man opened the pull request:
This PR adds psalm as a dev dependency to the plugin. It also updates the github action to also run the type check for PRs as well as adding all type fixes to all classes that is implemented.

This PR requires #84 to be reviewed first so that no changes are necessary for the Notification Banner class.

To test:

1. Fetch this branch `feature/psalm-dependency`.
2. Check you're running php8.2^
3. Run  `vendor/bin/psalm --error-level=1 --find-unused-psalm-suppress --no-cache`

snim2 commented on app/BlockController.php:
```diff
@@ -57,8 +60,10 @@ public function hasParent($displayName)
 		}
-
+		/** @var string */
 		$contents = file_get_contents($path);
```
Can this function fail in any way?

matthewpassmore replied:
Rather than reach for docblocks — L63 and L66 — to instruct Psalm, the code can be tightened up to check if `file_get_contents` is returning `false`, and `json_decode` is returning an array. This _should_ bring more certainty to return types.

william-man commented:
@matthewpassmore updated with the following:
- added realpath to normalise the path to the block.json (I think this is right). The function also checks whether the file exists so `file_exists() `has been removed and instead checks whether `false` is set instead.
- added an extra check for `file_get_contents` to guard against failures for getting the block.json content and replace the annotation for the variable.
- similar addition for `json_decode` where it checks for an array instead.

matthewpassmore commented:
You'll need to rebase this first, to remove Notification work.
